src/backend/common/deferred/__init__.py

```diff
@@ -38,6 +42,67 @@ def defer(
     queue.enqueue(task, url=_url, headers=headers, service=service)
 
 
+def enqueue(
+    *,
+    url: str,
+    headers: Dict[str, str] = {},
```

Comment:
We should really pass `headers` along here to `_run_enqueue`. It's also wild because like - I'm pretty sure `defer` just drops these headers as well (although like - what would it do with them?)